dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1559,6 +1558,8 @@ void InterpreterSelectQuery::executeFetchColumns(
             });
         }
     }
+
+    pipeline.setMaxThreads(max_streams);
```

Comment:
This is still quite not correct because of the code above
```c++
        if (max_streams > 1 && !is_remote)
            max_streams *= settings.max_streams_to_max_threads_ratio;	         
```
Here we increase the number of streams, but we don't want to increase the number of threads.

Reply:
Ok, fixed by setting it upper